core/chains/evm/logpoller/log_poller.go

```diff
@@ -415,6 +531,15 @@ func (lp *logPoller) pollAndSaveLogs(ctx context.Context, currentBlockNumber int
 	}
 }
 
+func (lp *logPoller) groupByEventID(logs []types.Log) map[EventID][]types.Log {
+	logsByEventID := make(map[EventID][]types.Log)
+	for _, log := range logs {
+		eventID := EventID{Address: log.Address, EventSig: log.Topics[0]} // TODO: could topics ever be len < 1?
```

Comment:
Do we need to address this comment?

Reply:
for all chainlink contracts topics >= 1, but I'll dig into whether thats possible in the evm in general (first topic being the event signature)

Reply:
is is indeed possible with an anonymous event, but the lp enforces you specify and event sig when adding to the filter, so we'd never have topics < 1. Added a comment to that effect.